packages/kbn-openapi-generator/src/template_service/templates/zod_operation_schema.handlebars

```diff
@@ -25,8 +26,14 @@ import {
   {{/if}}
   */
 {{/if}}
+{{#if (isCircularSchema @key)}}
+export type {{@key}} = {{> ts_type}};
+export type {{@key}}Input = {{> ts_input_type }};
+export const {{@key}}: z.ZodType<{{@key}}, ZodTypeDef, {{@key}}Input> = {{> zod_schema_item }};
```

Comment:
Can we reuse the same approach for all generated schemas, not only circular ones? Generated TS types provide much better DX than `infer`, IMO.